### ISSUE TITLE
ESLint doesn't validate schema for built-in rules

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13793]
**Tell us about your environment**

* **ESLint Version:** v7.12.0
* **Node Version:** v6.14.6
* **npm Version:** v12.18.4

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  rules: {
    "accessor-pairs": ["error", { foo: 1 }]
  }
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// anything, the config is important
```

```bash
npx eslint index.js
```

**What did you expect to happen?**

Configuration error for `accessor-pairs` rule.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors.

**Are you willing to submit a pull request to fix this bug?**

https://github.com/eslint/eslintrc/pull/14
